radicle-cli/src/commands/ls.rs

```diff
@@ -47,18 +47,37 @@ pub fn run(_options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
     let storage = &profile.storage;
     let mut table = term::Table::default();
 
-    storage.repositories()?.into_iter().for_each(|id| {
-        let Ok(repo) = storage.repository(id) else { return };
-        let Ok((_, head)) = repo.head() else { return };
-        let Ok(proj) = repo.project_of(profile.id()) else { return };
+    for id in storage.repositories()? {
+        let repo = match storage.repository(id) {
+            Ok(repo) => repo,
+            Err(e) => {
+                eprintln!("{}", e);
+                continue;
+            }
+        };
+        let head = match repo.head() {
+            Ok((_, head)) => head,
+            Err(e) => {
+                eprintln!("{}", e);
+                continue;
+            }
+        };
+        let proj = match repo.project_of(profile.id()) {
+            Ok(proj) => proj,
+            Err(e) => {
+                eprintln!("{}", e);
+                continue;
+            }
+        };
```

Comment:
Nah this needs more thought, it will be very annoying as-is if users don't know how to get rid of these errors.

Reply:
Could we have a  function `projects` that lists projects with `impl Iterator<Item = Result<ProjectInfo, Error>>`. In the `for_each` we could then `term::warn` that a project is broken if it's an error and `--verbose` is used.

Reply:
> Could we have a function `projects` that lists projects with `impl Iterator<Item = Result<ProjectInfo, Error>>`. In the `for_each` we could then `term::warn` that a project is broken if it's an error and `--verbose` is used.

This patch has been merged via Radicle already, so what you’re proposing would require a new one

Reply:
Cool, can you point me to the relevant patch, please?